nengo_spa/algebras/base.py

```diff
@@ -104,6 +104,38 @@ def bind(self, a, b):
         """
         raise NotImplementedError()
 
+    def fractional_bind(self, v, exponent):
```

Comment:
I'm wondering a bit what the best name for this method would be. Could there be algebras that support multiple auto-bindings, but only with integer exponents (in which case it wouldn't really be fractional binding)? If so, I could see this being implemented via the same method, but you wouldn't necessarily be able to use that algebra everywhere (depending on whether non-integer exponents are used).

`pow` would be analogous to the `pow` function which this sort of binding is, but that might be not-obvious. What do you think of `autobind`?

Reply:
Plate (1995; section 3.6.5) calls this "the convolutive power" with "fractional exponents". The "fractional" term here is also consistent with [fractional differentiation](https://en.wikipedia.org/wiki/Fractional_calculus). It is a bit of a misnomer because "fractional" isn't limited to the rationals. However this naming appears to be used elsewhere as well in the same context (e.g., [scipy.linalg.fractional_matrix_power](https://docs.scipy.org/doc/scipy-0.15.1/reference/generated/scipy.linalg.fractional_matrix_power.html)), so to us it seemed reasonable to bring in the fractional term here as well, since the emphasis is on the same generatlization. This is also the naming we went with in the cogsci papers. 

> Could there be algebras that support multiple auto-bindings, but only with integer exponents (in which case it wouldn't really be fractional binding)?

Probably. Binary Spatter Codes might be one such instance.

> I could see this being implemented via the same method, but you wouldn't necessarily be able to use that algebra everywhere (depending on whether non-integer exponents are used).

If we get there, I could see a separate method being implemented for the integer case for at least two reasons. One, because it seems likely we'd want different neural implementations for the two methods, and two because we might later want to allow the fractional exponent to be provided on-the-fly to the network by some other ensemble (for the fractional binding case, and likely not for the integer case).

Reply:
These are some good points for sure. Especially if this naming has been used in published literature before. Thinking a bit more about this, I might be less worried about the "fractional" part of the term, but more about the "binding" part. Binding is analogous to multiplication. But neither Plate nor in the matrix example are we talking about "fractional multiplication"; it is a "fractional **power**". Thus, it seems to mewe're missing a word analogous to "power" for "binding" that should be used here. Also, "binding" somewhat implies to me that something is bound to something else, i.e. there are two Semantic Pointers involved (but maybe others don't have that intuition).

If we want to provide separate implementations for integer and fractional powers (which might be reasonable), it raises another interesting question: There is only one `**` operator. Which power would it represent? How would we handle the other variant if there were neural implementations (e.g. what would `state ** scalar` mean, and how to get the alternate implementation).

Just want to make sure, the API and names are well thought out. Once it's in a release, it will be harder to change. :)